HousingSearchApi.Tests/V2/E2ETests/TenureSearchTests.cs

```diff
@@ -0,0 +1,253 @@
+        var firstResult = root.GetProperty("results").GetProperty("tenures")[0];
+        firstResult.GetProperty("id").GetString().Should().Be(expectedReturnedId);
+    }
+
+    # endregion
+
+    # region Person
+
+    [Fact]
+    public async Task SearchPerson_Name()
+    {
+        const int attempts = 10;
+        const int minSuccessCount = 9;
+
+        var successCount = await RunWithScore(attempts, async () =>
+        {
+            // Arrange
+            var tenure = RandomItem();
+            var expectedReturnedId = tenure.GetProperty("id").GetString();
+            var householdMembers = tenure.GetProperty("householdMembers");
+            var randomMember = householdMembers[new Random().Next(householdMembers.GetArrayLength())];
+            var searchText = randomMember.GetProperty("fullName").GetString();
+            var request = CreateSearchRequest(searchText);
+
+            // Act
+            var response = await _httpClient.SendAsync(request);
+
+            // Assert
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            var root = GetResponseRootElement(response);
+            root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
+            var firstResult = root.GetProperty("results").GetProperty("tenures")[0];
+            firstResult.GetProperty("id").GetString().Should().Be(expectedReturnedId);
+        });
+
+        successCount.Should().BeGreaterThanOrEqualTo(minSuccessCount);
+    }
+
+    [Fact]
+    public async Task SearchPerson_NameCutoff()
+    {
+        const int attempts = 10;
+        const int minSuccessCount = 9;
+
+        var successCount = await RunWithScore(attempts, async () =>
+        {
+            // Arrange
+            var tenure = RandomItem();
+            var expectedReturnedId = tenure.GetProperty("id").GetString();
+            var householdMembers = tenure.GetProperty("householdMembers");
+            var randomMember = householdMembers[new Random().Next(householdMembers.GetArrayLength())];
+            var memberName = randomMember.GetProperty("fullName").GetString();
+            var nameParts = memberName.Split(' ');
+            // Remove a character from a random name part
+            nameParts[new Random().Next(nameParts.Length)] = nameParts[new Random().Next(nameParts.Length)][..^1];
+            var searchText = string.Join(" ", nameParts);
+
+            var request = CreateSearchRequest(searchText);
+
+            // Act
+            var response = await _httpClient.SendAsync(request);
+
+            // Assert
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            var root = GetResponseRootElement(response);
+            root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
+            var firstResult = root.GetProperty("results").GetProperty("tenures")[0];
+            firstResult.GetProperty("id").GetString().Should().Be(expectedReturnedId);
+        });
+
```

Comment:
Is there any way to abstract some of this test setup? There is a lot of duplication